src/main/java/io/sealights/plugins/sealightsjenkins/buildsteps/cli/executors/AbstractCommandExecutor.java

```diff
@@ -21,71 +20,42 @@
 
     protected Logger logger;
     protected BaseCommandArguments baseArgs;
-    private Runtime runtime;
+    private ProcessExecutor processExecutor;
 
     public AbstractCommandExecutor(Logger logger, BaseCommandArguments baseArgs) {
         this.logger = logger;
         this.baseArgs = baseArgs;
-        this.runtime = Runtime.getRuntime();
+        this.processExecutor = new ProcessExecutor();
     }
 
     public boolean execute() {
         try {
-            String[] execCommand = createExecutionCommand();
-
-            // Run a java app in a separate system process
-            logger.info("About to execute command: " + Arrays.toString(prettifyToken(execCommand)));
-
-            Process process = runtime.exec(execCommand);
-            process.waitFor();
-
-            printStreams(process);
-
-            if (process.exitValue() == 0) {
-                return true;
-            }
-
+            ProcessResult result = processExecutor.command(createExecutionCommand()).readOutput(true).execute();
+            String output = result.outputUTF8();
+            logger.info(output);
```

Comment:
I would like to see in the demo if that's easy to understand that this is the log of the step. Maybe it is worth prefixing it.
Also, make sure to test failure to reach the server and verify that the log is printed correctly.